e2etest/l2tests/l2.go

```diff
@@ -514,21 +514,39 @@ var _ = ginkgo.Describe("L2", func() {
 				if err != nil {
 					return err
 				}
-				err = metrics.ValidateGaugeValue(0, "metallb_allocator_addresses_in_use_total", map[string]string{"pool": poolName}, controllerMetrics)
-				if err != nil {
-					return err
+				type Cases []struct {
+					name   string
+					expect int64
 				}
-				err = metrics.ValidateGaugeValue(int(poolCount), "metallb_allocator_addresses_total", map[string]string{"pool": poolName}, controllerMetrics)
-				if err != nil {
-					return err
-				}
-				err = metrics.ValidateOnPrometheus(promPod, fmt.Sprintf(`metallb_allocator_addresses_in_use_total{pool="%s"} == 0`, poolName), metrics.There)
-				if err != nil {
-					return err
+				cases := Cases{
```

Comment:
nit: given it's local, you can use an anonymous struct here (and no need to have it exported anyway).

Something like cases := []struct{name,
                                                          expect}{
                                            }

Reply:
nice trick.